package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -174,132 +167,319 @@ class WaterBridgeAnalysis_OtherFF(WaterBridgeAnalysis):
+Use count_by_type
+---------------------------
 
-Alternatively, :meth:`~WaterBridgeAnalysis.count_by_type` can also be used to
+To calculate the percentage, we can use the :meth:`~WaterBridgeAnalysis.count_by_type` to
 generate the frequence of all water bridges in the simulation. ::
 
   w.count_by_type()
 
 Returns ::
 
-  [(0, 3, 'ARG', 1, 'O', 'ASP', 3, 'O', 0.5)]
+  [(0, 3, 'ARG', 1, 'O', 'ASP', 3, 'OD1', 0.5),
+   (0, 4, 'ARG', 1, 'O', 'ASP', 3, 'OD2', 0.5),]
 
-For further data analysis, it is convenient to process the
-:attr:`~WaterBridgeAnalysis.timeseries` data into a normalized table with the
-:meth:`~WaterBridgeAnalysis.generate_table` method, which creates a new data
-structure :attr:`WaterBridgeAnalysis.table` that contains one row for each
-observation of a hydrogen bond::
+You might think that the OD1 and OD2 are the same oxygen and the aspartate has just flipped
+and thus, they should be counted as the same type of water bridge. The type of the water
+bridge can be customised by supplying an analsysis function to
+:meth:`~WaterBridgeAnalysis.count_by_type`.  ::
 
-  w.generate_table()
+  def analysis(current, output):
+      '''This function defines how the type of water bridge should be specified.
 
-This table can then be easily turned into, e.g., a `pandas.DataFrame`_, and
-further analyzed::
+        Parameters
+        ----------
+        current : list
+            The current water bridge being analysed is a list of hydrogen bonds from
+            selection 1 to selection 2.
+        output : dict
+            A dictionary where the key is the type of the water bridge and the value
+            is the number of this type of water bridge.
+      '''
+
+      # decompose the first hydrogen bond.
+      s1_index, to_index, (s1_resname, s1_resid, s1_name),
+      (to_resname, to_resid, to_name), dist, angle = current[0]
+      # decompose the last hydrogen bond.
+      from_index, s2_index, (from_resname, from_resid, from_name),
+      (s2_resname, s2_resid, s2_name), dist, angle = current[-1]
+      # if the residue name is ASP and the atom name is OD2 or OD1,
+      # the atom name is changed to OD
+      if s2_resname == 'ASP' and (s2_name == 'OD1' or s2_name == 'OD2'):
+          s2_name = 'OD'
+      # setting up the key which defines this type of water bridge.
+      key = (s1_resname, s1_resid, s1_name, s2_resname, s2_resid, s2_name)
+      # The number of this type of water bridge is incremented by 1.
+      output[key] += 1
+
+  w.count_by_type(analysis_func=analysis)
 
-  import pandas as pd
-  df = pd.DataFrame.from_records(w.table)
+Returns ::
 
+  [(('ARG', 1, 'O', 'ASP', 3, 'OD'), 1.0),]
 
-.. _pandas.DataFrame: http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.html
+Note that the result is arranged in the format of (key, proportion of time). When no
+custom analysis function is supplied, the key is expended for backward compatibility.
```

Comment:
What "backward compatibility"? Compatible to what? 

How is the analysis function being used? This is not clear here. If you explain it below, say so and make a link, or restructure.

Reply:
@orbeckst Compatible to the function of the same name in hydrogen bond analysis.
The aim of this is that if the same command is executed in hydrogen bond analysis and water bridge analysis they will return the same result.

Reply:
- fix expended -> expanded (?)
- state what it is backward compatible to

Reply:
More explanation has been added.